File: sdkwork-rtc-sdk-swift/Sources/RtcSdk/RtcStandardContract.swift
public enum RtcStandardContract {
    public static let symbol = "RtcStandardContract"
}

public protocol RtcProviderDriver {
    var providerKey: String { get }
}

public protocol RtcDriverManager {
    func resolveDriver(providerKey: String)
}

public protocol RtcDataSource {
    func createClient() async throws
}

public protocol RtcClient {
    func join() async throws
    func leave() async throws
    func publish(trackId: String) async throws
    func unpublish(trackId: String) async throws
    func muteAudio(muted: Bool) async throws
    func muteVideo(muted: Bool) async throws
    func unwrap() -> Any?
}

public protocol RtcRuntimeController {
    func join() async throws
    func leave() async throws
    func publish(trackId: String) async throws
    func unpublish(trackId: String) async throws
    func muteAudio(muted: Bool) async throws
    func muteVideo(muted: Bool) async throws
}

File: sdkwork-rtc-sdk-csharp/src/SDKWork.Rtc.Sdk/RtcStandardContract.cs
namespace Sdkwork.Rtc.Sdk;

public static class RtcStandardContract
{
    public const string Symbol = "RtcStandardContract";
}

public interface RtcProviderDriver<TNativeClient>
{
    string ProviderKey { get; }

    RtcClient<TNativeClient> CreateClient();
}

public interface RtcDriverManager<TNativeClient>
{
    RtcProviderDriver<TNativeClient> ResolveDriver(string providerKey);
}

public interface RtcDataSource<TNativeClient>
{
    RtcClient<TNativeClient> CreateClient();
}

public interface RtcClient<TNativeClient>
{
    void Join();

    void Leave();

    void Publish(string trackId);

    void Unpublish(string trackId);

    void MuteAudio(bool muted);

    void MuteVideo(bool muted);

    TNativeClient? Unwrap();
}

public interface RtcRuntimeController<TNativeClient>
{
    void Join();

    void Leave();

    void Publish(string trackId);

    void Unpublish(string trackId);

    void MuteAudio(bool muted);

    void MuteVideo(bool muted);
}
